schedule/models/events.py

```diff
@@ -59,8 +62,91 @@ def __unicode__(self):
     def get_absolute_url(self):
         return reverse('event', args=[self.id])
 
-    def get_occurrences(self, start, end, skip_booster=False):
+    def get_recent_start_validated(self):
+        """
+         Unserializes and validates self.recent_start. Returns the dictionary
+         represented by self.recent_start if all goes well, or None if there
+         is a validation error.
+         @return recent_start converted to object if it's valid given current
+         state of event; None if it doesn't exist or is invalid.
+        """
+        if self.recent_start == None:
+            return None
+
+        recent_start_data = pickle.loads(str(self.recent_start))
+
+        if (recent_start_data['event']['start'] == self.start and
+              recent_start_data['event']['rule'] == self.rule):
+            return recent_start_data
+        else:
+            return None
+
+
+    @property
+    def recent_occurrence_start(self):
         """
+        A lazy loaded property that ensures we unpickle to get the recent
+        occurrence start only once. Use similar pattern for other properties
+        :return:
+        """
+        if not hasattr(self, '__recent_start'):
+            setattr(self, '__recent_start', self.get_recent_start_validated())
+        recent_start = getattr(self, '__recent_start', None)
+        if (recent_start is not None
+            and 'occurrence' in recent_start
+            and 'start' in recent_start['occurrence']):
+            return recent_start['occurrence']['start']
+        else:
+            return None
+
+
+    def build_recent_start_payload(self, occurrence):
```

Comment:
This will get called in the async task method in vishnu backend that is responsible for writing to recent_events. You can see a rough version here: https://github.com/AltSchool/vishnu-backend/blob/faisal/osfifone-1100_event_memoizing/vishnu/tasks.py#L380